### PR TITLE
feature/1675 - Add MCO

### LINKED ISSUE #1675: Add KMCO - Orlando Int'l
Add MCO.

### PR DESCRIPTION
**Pull request by MichaelDMcCracken**
Resolves #1675 

The purpose of this pull request is to add MCO.

### CODE REVIEW

**Comment by Fechulo** [comment]
Hey @MichaelDMcCracken 

You requested a review for this around 7 months ago, sorry for the delay. It seems this is still very much a work in progress, since it's still missing a lot. Have a look at the [airport file standards document](https://github.com/openscope/openscope/blob/develop/documentation/airport-file-standards.md) for more info. Let me know if you have any questions / need any help.

**Comment by MichaelDMcCracken** [comment]
Ready for review, but I need someone to update it to the correct version. Thanks.

File: assets/airports/kmco.json [hunk 1]
```diff
@@ -0,0 +1,6135 @@
...
+                ["SRQ*"],
+                ["SZW*"]
+            ]
+        }
+    },
+    "stars": {
+        "ALYNA1": {
+            "icao": "ALYNA1",
+            "name": "Alayna One",
+            "entryPoints": {
```
Comment (Fechulo) [inline assets/airports/kmco.json]: Missing BEAUX and SURFR transitions

File: assets/airports/kmco.json [hunk 2]
```diff
@@ -0,0 +1,6135 @@
...
+                ["PHK*", "BAIRN"]
+            ]
+        },
+        "GRNCH2": {
+            "icao": "GRNCH2",
+            "name": "Grinch Two",
+            "entryPoints": {
+                "ADLYN": ["ADLYN", "ELITE", ["MAYYR", "A240+"]],
+                "CATEE": ["CATEE", "ELITE", ["MAYYR", "A240+"]],
+                "COAXE": ["COAXE"]
```
Comment (Fechulo) [inline assets/airports/kmco.json]: COAXE should be moved into the body since all transitions pass through it

File: assets/airports/kmco.json [hunk 3]
```diff
@@ -0,0 +1,6135 @@
...
+            "draw": [
+                ["ADLYN*", "ELITE", "MAYYR", "COAXE*", "GRTNG", "GRNCH", "BUGGZ", "BLAYK", "BUNIE", "YAGGO"],
+                ["CATEE*", "ELITE"],
+                ["BLAYK", "WHOVL", "HARAM", "DORII", "BRICE"]
+            ]  
+        },
+        "JAFAR2": {
+            "icao": "JAFAR2",
+            "name": "Jafar Two",
+            "entryPoints": {
```
Comment (Fechulo) [inline assets/airports/kmco.json]: Missing BRKWL transition

File: assets/airports/kmco.json [hunk 4]
```diff
@@ -0,0 +1,6135 @@
...
+               ["COEDY*", "BUULL", "HUNKR", "DOWNO", "BRKWL", "JAFAR", "CRPIT", "LAMMP", "WISHH", "HARAM", "DORII", "BRICE"],
+               ["ILEXX*", "HUNKR"],
+               ["BITTE*", "BRKWL"],
+               ["LAMMP", "YAGGO"]
+            ]
+        },
+        "JOKRS2": {
+            "icao": "JOKRS2",
+            "name": "Jokers Two",
+            "entryPoints": {
```
Comment (Fechulo) [inline assets/airports/kmco.json]: Missing FAZES and MOANS transitions

File: assets/airports/kmco.json [hunk 5]
```diff
@@ -0,0 +1,6135 @@
...
+            "draw": [
+                ["COREA*", "SIMIE", "FAZES", "MOANS", "JOKRS", "TAYYE", "ARHIP", "GIRAF", "DVOLA", "LYENS"],
+                ["FERAL*", "SIMIE"],
+                ["JOKRS", "LORRA"]
+            ]
+        },
+        "LEESE3": {
+            "icao": "LEESE3",
+            "name": "Leese Three",
+            "entryPoints": {
```
Comment (Fechulo) [inline assets/airports/kmco.json]: Missing OCF transition

File: assets/airports/kmco.json [hunk 6]
```diff
@@ -0,0 +1,6135 @@
...
+              "OTK": ["OTK"],
+              "SAV": ["SAV", "SHEMP"]
+            },
+            "body": ["OCF", "ALADN"],
+            "rwy": {
+              "KMCO17L": [["SHIMM", "A110|S250"], "LEESE"],
```
Comment (Fechulo) [inline assets/airports/kmco.json]: There are no restrictions at SHIMM as far as I can see so everything in the rwy section can be moved to the body.
Also after LEESE, missing ORL

File: assets/airports/kmco.json [hunk 7]
```diff
@@ -0,0 +1,6135 @@
...
+            },
+            "draw": [
+                ["OTK*", "OCF", "ALADN", "SHIMM", "LEESE"],
+                ["SAV*", "SHEMP", "OCF"]
+            ]
+        },
+        "MINEE5": {
+            "icao": "MINEE5",
+            "name": "Minnie Five",
+            "entryPoints": {
```
Comment (Fechulo) [inline assets/airports/kmco.json]: Missing PIE transition

File: assets/airports/kmco.json [hunk 8]
```diff
@@ -0,0 +1,6135 @@
...
+              "BOXKR": ["BOXKR", "GUMMY", "LAL", ["ANDRO", "A130"]],
```
Comment (Fechulo) [inline assets/airports/kmco.json]: Missing SYKES after BOXKR

File: assets/airports/kmco.json [hunk 9]
```diff
@@ -0,0 +1,6135 @@
...
+            "draw": [
+                ["BOXKR*", "SYKES", "GUMMY", "LAL", "ANDRO", "MINEE", "UNITY", "ORL"],
+                ["LBV*", "DOWNN", "MOANS", "MINEE"],
+                ["RSW*", "DOWNN"]
+            ]
+        },
+        "PRICY1": {
+            "icao": "PRICY1",
+            "name": "Pricy One",
+            "entryPoints": {
```
Comment (Fechulo) [inline assets/airports/kmco.json]: Missing BULKS and CHIVE transitions

File: assets/airports/kmco.json [hunk 10]
```diff
@@ -0,0 +1,6135 @@
...
+              "BAYNE": ["BAYNE"],
+              "CHRGE": ["CHRGE", "DAAWG", ["CHIVE", "A240+|A260-"], ["DECKS", "A210-"]],
```
Comment (Fechulo) [inline assets/airports/kmco.json]: DECKS should be at **or above** FL210

File: assets/airports/kmco.json [hunk 11]
```diff
@@ -0,0 +1,6135 @@
...
+              "DOZES": [["DOZES", "A230"], ["QUAYL", "A210+"]]
+            },
+            "body": [["BULKS", "A190+|A200-|S280"], ["PRICY", "A130+"], "BOCCE", "JUMBA"],
+            "rwy": {
+              "KMCO17L": [["TAYYE", "A100+|A120-|S250"], "ARHIP", ["GIRAF", "A80+"], "DVOLA", ["^LYENS", "A50|S220"], "#006"],
```
Comment (Fechulo) [inline assets/airports/kmco.json]: LYENS should be at 210kts

File: assets/airports/airportLoadList.js [hunk 12]
```diff
@@ -276,6 +276,12 @@ window.AIRPORT_LOAD_LIST = (function() { // eslint-disable-line wrap-iife
             name: 'Kansas City International Airport',
             premium: false
         },
+        {
+            icao: 'kmco',
+            level: 'beginner',
```
Comment (Fechulo) [inline assets/airports/airportLoadList.js:281]: With a current rate of 74.8 acph, this should be changed to 'hard'

File: documentation/airport-guides/kmco.md [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+
+## Airspace
+The MCO airspace is fairly large and covers multiple smaller airports from the surface to 16,000'.
+
+## Procedures
+MCO is served primarily by RNAV departure and arrival procedures. RNAV STARs are highly automated and require little ATC intervention. In North Flow (35/36s) RNAV SIDs are RNAV-Off-the-Ground to the en route transition. In South Flow (17/18s), RNAV SIDs begin with a coded heading upon departure wihch requires radar vectors to the first fix.
+
+## Additional Resources
+<a href="https://zjxartcc.org/media/doc/ZJX-1.D_Jacksonville_ARTCC_MCO_ATCT_SOP_pC6lZjs.pdf" target="_blank">Vatsim ZJX — Orlando ATCT Procedures</a>
```
Comment (Fechulo) [inline documentation/airport-guides/kmco.md:15]: This document might be more useful:
https://zjxartcc.org/media/doc/ZJX-10.G_Jacksonville_ARTCC_F11_SOP_FDlIm85.pdf

File: documentation/airport-guides/kmco.md [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+
+The airport serves as a hub for Silver Airways, an operating base for JetBlue, Southwest Airlines and Spirit Airlines, as well as a focus city for Frontier Airlines. Southwest is the airport's largest carrier by passengers carried. The airport is also a major international gateway for the mid-Florida region, with over 850 daily flights on 44 airlines. The airport also serves 135 domestic and international destinations. At 12,600 acres (5,100 ha), MCO is one of the largest commercial airports in terms of land area in the United States. In addition, the airport is home to a maintenance base for United Airlines. The airport was also a hub for Delta Air Lines until 2007.(Wikipedia)
+
+## Airspace
+The MCO airspace is fairly large and covers multiple smaller airports from the surface to 16,000'.
```
Comment (Fechulo) [inline documentation/airport-guides/kmco.md:9]: Officially the F11 TRACON

**Review by Fechulo — CHANGES_REQUESTED** [review]

**Comment by Fechulo** [comment]
@MichaelDMcCracken There are also some restricted areas to the North, South and East of MCO. They might be outside the TRACON but are still significant and should be included

File: assets/airports/kmco.json (hunk 3 above)
Comment (MichaelDMcCracken) [reply assets/airports/kmco.json]: This seems contradictory to the statement above about the COAXE fix. All transitions pass through BRKWL. So do they need to go in the body or entryPoints?

**Comment by Fechulo** [comment]
@MichaelDMcCracken See also line 442 of kmco.json:
FSHUN is misspelled in the body and causes sim to crash

**Comment by MichaelDMcCracken** [comment]
image

@Fechulo 
One issue I've noticed is that the aircraft on departure do not fly the exact assigned heading from the SID. It is only 1 degree off. They are set to fly 176° (from 17R) and 191° (from 18L), but fly 177° and 192° respectively. Even though this is 15° apart, the sim is still giving the red rings. What's the deal?

**Comment by Fechulo** [comment]
@erikquinn might be able to answer that question better

**Comment by MichaelDMcCracken** [comment]
> There are also some restricted areas to the North, South and East of MCO. They might be outside the TRACON but are still significant and should be included
@Fechulo 
I added the north and south restricted areas. I did not add the restricted areas to the east at Cape Canaveral because they are only active during rocket launch windows and the MZULO departure goes right through the middle. Also added the Disney World TFR. Everything should be ready to go!

**Comment by erikquinn** [comment]
The heading discrepancy is likely working correctly. The heading in the chart is a ground track to fly, and not really a heading (ie, they should apply wine correction to follow a 166 track off 17L (from your example). Given the wind of 220@7, they will need to crab slightly to the right to counteract the right-to-left crosswind they experience. One degree for that sounds about right.

As for the rings when their headings are 15 degrees apart, I suspect that to be an issue of rounding-- where they're like 14.846294600479 degrees divergent. A few things to fix there, which could go in a new issue. We could (a) allow it if `> minDivergence -1` to account for rounding errors, and (b) allow 10 degrees divergence when on an RNAV SID. The latter might be tough, but the former is easy. Would solve your issue while also punishing someone if they happened to assign 010 and 024.

**Comment by MichaelDMcCracken** [comment]
> The heading discrepancy is likely working correctly. The heading in the chart is a ground track to fly, and not really a heading (ie, they should apply wind correction to follow a 166 track off 17L (from your example). Given the wind of 220@7, they will need to crab slightly to the right to counteract the right-to-left crosswind they experience. One degree for that sounds about right.
> 
> As for the rings when their headings are 15 degrees apart, I suspect that to be an issue of rounding-- where they're like 14.846294600479 degrees divergent. A few things to fix there, which could go in a new issue. We could (a) allow it if `> minDivergence -1` to account for rounding errors, and (b) allow 10 degrees divergence when on an RNAV SID. The latter might be tough, but the former is easy. Would solve your issue while also punishing someone if they happened to assign 010 and 024.

Copy. Wasn't sure if the `sh` command gave you a _response_ of the ground track or actual heading. 

My guess is that it is also a rounding error, because if I tell the aircraft on the 177° heading to `t l 1` it instantly corrects the issue.

**Review by Fechulo — APPROVED** [review]

**Pull request merged**